### ISSUE TITLE
Was the selection of the GPL license rather than something like the MIT license deliberate? 

### ISSUE THREAD

**Issue by Carbaholic** [arklumpus/VectSharp#11]
I would love to use your library in an app I'm writing at work. 

Under the GPL license, once I distribute my app I would have to make my source available under the GPL license. If your choice of the GPL license was deliberate, I'll have to move along and try to find another library. If it wasn't, would you mind changing your license to something less restrictive such as MIT? 

Thanks!

**Comment by arklumpus** [comment]
Hi! I'm glad to hear you're interested in my library!

I've actually been thinking for some time about changing the licence of my libraries to LGPL, would that be enough for you? It should remove the requirement to release the source code of your software (of course, unless you modify the source of VectSharp itself).

Note however that VectSharp.Raster and VectSharp.MuPDFUtils have to be released as AGPL (because they use Artifex's MuPDF). Although at some point I would like to move to Skia instead of MuPDF...

**Comment by Carbaholic** [comment]
LGPL would definitely work for me :)

**Comment by arklumpus** [comment]
Perfect! I'll close this issue and create the new release as soon as I have time to change all the licence notices.

**Comment by dstj** [comment]
@arklumpus @Carbaholic 

I'm no expert, but doesn't [LGPL 3.0](https://tldrlegal.com/license/gnu-lesser-general-public-license-v3-(lgpl-3)) also requires you to disclosure source code of your app if it within the same executable?

> Disclose Source
> If you distribute this library in an executable, you must make the source available for 3 years.

I'd also possibly like to use it in my commercial application to generate SVG file on the fly, but obviously, I do not want to disclose ALL my source code :)

**Comment by arklumpus** [comment]
IANAL either, but I don't think that's the case (at least, not exactly). Basically, you have two options:

1. Distribute your program and the VectSharp DLL as separate files (in the sense that when the user installs your program, they get a folder on their computer that contains `YourProgram.exe` and `VectSharp.dll` as separate files):
    * In this case, you only need to provide the source code for VectSharp - e.g. by linking to this repository.

2. Distribute your program and VectSharp as a single file (in this case, when the user installs the program they only get `YourProgram.exe`):
    * In this case, you don't really have to provide the source code for your program either; _however_, you must provide instructions on how to recompile the program including a different version of VectSharp (which is not necessary in case 1, because you can simply replace the DLL file and everything should work). Of course, this might not be possible without disclosing all the source code, but it might be.

I would expect most .NET applications to be released as in option 1, so it shouldn't be an issue... I have tried using the "Publish as a single file" option, but it seems to always produce more than one file anyways (especially when native libraries are involved).

TL;DR: if you don't use the "Publish as a single file" option, you should be fine. You need to include the appropriate licence notices, but that's no different from MIT.

More detailed information [here](https://copyleft.org/guide/), see especially sections 10 and 11.

**Comment by AlexanderMagdysyuk** [comment]
Hi

So you need to remove/extract 14 standard fonts?

Thanks.

**Comment by arklumpus** [comment]
Now, _that_ is a good question... It turns out that there was a reason why the library is released as GPL 😂 

Currently, the fonts (which are released under GPL) are embedded as resources within the VectSharp DLL, which will not be possible when VectSharp becomes LGPL.

The easiest solution would be to replace the fonts with metrics-compatible alternatives that are released under a licence that is compatible with LGPL; for example, the Chrome OS fonts (Arimo, Tinos, Cousine) are released under Apache 2.0, which should be compatible with LGPL.

Another possibility would be to create a mechanism to load a custom "font library" by specifying font files for the standard fonts, which would replace the default fonts. In this way, you could use fonts under whatever licence (GPL, SIL OFL, a commercial licence, etc.), as long as it is compatible with the licence you choose for your program. For example, the SIL OFL is not compatible with the LGPL, which means I cannot embed the Liberation fonts in VectSharp; however using this mechanism you could still distribute the font files with your application while complying with the licences. I could also create a GPL NuGet package including the fonts that are currently used in VectSharp, which would allow people releasing GPL'd software to keep the current behaviour.

**Comment by AlexanderMagdysyuk** [comment]
Found Apache 2.0 fonts: https://github.com/levigo/documentplatform-standard14-fonts (except Symbol) and metrics for all fonts: https://github.com/Hopding/standard-fonts

**Comment by arklumpus** [comment]
Yeah, those are the Chrome OS fonts I was talking about, Symbol is available from here: http://commondatastorage.googleapis.com/chromeos-localmirror/distfiles/croscorefonts-1.23.0.tar.gz

**Comment by arklumpus** [comment]
I have finally managed to create the LGPL release! I'm closing this now, but let me know if you have any other issues!
